pkg/networkservice/nsmonitor/client_test.go

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) 2022 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package nsmonitor_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/common"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/begin"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/count"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/nsmonitor"
+)
+
+const (
+	testInode1 = "inode://4/4026534206"
+	testInode2 = "inode://4/4026534149"
+)
+
+type testMonitor struct {
+	inodes                     []string
+	watchShouldCloseMonitoing  bool
+	watchShouldCloseConnection bool
+	mutex                      sync.Mutex
+}
+
+func (m *testMonitor) Watch(ctx context.Context, inodeURL string) <-chan struct{} {
+	m.mutex.Lock()
+	m.inodes = append(m.inodes, inodeURL)
+	m.mutex.Unlock()
+
+	result := make(chan struct{}, 1)
+	if m.watchShouldCloseConnection {
+		result <- struct{}{}
+		close(result)
+	} else if m.watchShouldCloseMonitoing {
+		close(result)
+	}
+	return result
+}
+
+func (m *testMonitor) requireInodes(t *testing.T, inodeURLs []string) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	require.Equal(t, m.inodes, inodeURLs)
+}
+
+func Test_Client_DontFailWhenNoInode(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	monitor := testMonitor{}
+	supplyMonitor := func(c context.Context) nsmonitor.Monitor { return &monitor }
+
+	client := chain.NewNetworkServiceClient(
+		metadata.NewClient(),
+		begin.NewClient(),
+		nsmonitor.NewClient(ctx, nsmonitor.WithSupplyMonitor(supplyMonitor)),
+	)
+
+	// no inodeURL in parameters
+	_, err := client.Request(ctx, &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Id: uuid.New().String(),
+			Mechanism: &networkservice.Mechanism{
+				Parameters: map[string]string{},
+			},
+		},
+	})
+	require.NoError(t, err)
+	require.Empty(t, monitor.inodes)
```

Comment:
Should we use requireInodes?
```suggestion
	requireInodes(t, monitor.inodes)
```